lib/discourse_activity_pub/json_ld.rb

```diff
@@ -82,6 +82,21 @@ def public_collection_id
       PUBLIC_COLLECTION_IDS.first
     end
 
+    def resolve_icon_url(value)
+      return nil if value.nil?
+      return value if value.is_a?(String)
+      return value['url'] if value.is_a?(Hash)
+      return value.first['url'] if value.is_a?(Array)
```

Comment:
Just curious, why do we need to support so many formats for the icon url (string, hash, array)?

Reply:
Because it can be any of those, i.e. a simple url, a hash containing a url, or an array of hashes containing urls. ActivityStreams is quite liberal in its types for attributes. See https://www.w3.org/TR/activitystreams-vocabulary/#dfn-icon